ci/scripts/pre-unit-test.sh

```diff
@@ -6,6 +6,9 @@ set -euo pipefail
 source ci/scripts/common.env.sh
 
 echo "--- Run clippy check"
+cargo clippy --locked -- -D warnings
```

Comment:
Also add it to risedev? Otherwise it might be confusing if `risedev c` passed but ci failed

Reply:
Could I ask what is the purpose of adding this? Is it to test without sync_point? Or make the check run faster?

Reply:
> Also add it to risedev?

This will make the checking slower and I'm not sure whether it is worth it. 🥵

> Could I ask what is the purpose of adding this?

For example, there're two unused imports in `hummock/mod.rs` shown as warnings when we build with `risedev d`, but cannot be checked with `--all-targets --all-features`.